### DISCUSSION TITLE
lolin S3 pro with ILI9341

### DISCUSSION THREAD

**Discussion by ced31esp** [Bodmer/TFT_eSPI#3431] [Compatible displays and setup files]
Hello,

After years of using a lolin D32 pro, I decided to switch with a lolin S3 Pro . Unfortunately I can't make my ILI9341 working . Could you help me please. I changed the setup 70B like this : 
                    // Typical board default pins - change to match your board
#define TFT_CS   48 //     10 or 34 (FSPI CS0) 
#define TFT_MOSI 11 //     11 or 35 (FSPI D)
#define TFT_SCLK 12 //     12 or 36 (FSPI CLK)
#define TFT_MISO 13 //     13 or 37 (FSPI Q)

#define TFT_DC    47
#define TFT_RST   21

It's not working. 

It's always rebooting 
ELF file SHA256: 38e5d98b367089f5

Rebooting...
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x403799da
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3818,len:0x508
load:0x403c9700,len:0x4
load:0x403c9704,len:0xad0
load:0x403cc700,len:0x29e4
entry 0x403c9880
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4200507b  PS      : 0x00060430  A0      : 0x82005150  A1      : 0x3fcebce0  
A2      : 0x00000010  A3      : 0x00000000  A4      : 0x60004000  A5      : 0x0000000b  
A6      : 0x000000ff  A7      : 0x00000008  A8      : 0x08000000  A9      : 0x3fcebcb0  
A10     : 0x3fc96cd8  A11     : 0x00000001  A12     : 0x02625a00  A13     : 0xffffffff  
A14     : 0x00000031  A15     : 0x00000000  SAR     : 0x00000015  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000010  LBEG    : 0x420077c4  LEND    : 0x4200780b  LCOUNT  : 0x00000003  

My arduino version is 1.8.19 . 

My program is working with a simple SD test program . 

Maybe I'm using the wrong user setup file ? 

Could someone give me his configuration with lolin S3pro and ILI9341 . 
Thanks a lot.

**Comment by elgerg** [comment]
Is it this one?
https://www.wemos.cc/en/latest/s3/s3_pro.html

If so make note of the flash SPI mode:
16MB Flash (Quad SPI)

This is QSPI (or QIO) but you are using DSPI (mode:DIO, clock div:1).

Try changing this and it might just work.

**Reply by ced31esp** [reply]
![Capture d’écran 2024-08-05 à 14 24 26](https://github.com/user-attachments/assets/f653872d-599b-4349-a70b-ddc370a1ca72)


Thanks for your answer
I don't have lots of possibilities .
Where should I change your setting ? 
Thanks

**Reply by elgerg** [reply]
Ok, thats strange, can you try a generic S3 board and see if you can change that setting?

If thats not working you might have to raise a ticket/issue with WiMo.

**Reply by ced31esp** [reply]
I tried diferrent board on arduino but I can't change that setting too :-(

**Reply by ced31esp** [reply]
if anyone has already used a tft ili9341 with a lolin  S3 pro, I appreciate your advice.

**Reply by elgerg** [reply]
Did you try a basic blink sketch just so you are ruling out any TFP_eSPI stuff?

It would be useful to determine if it is the flash mode or not.

If it crashes with just a blink sketch then you will want to get it working before jumping into this lib.

**Comment by ced31esp** [comment]
Hello sorry it was written in my first message but lost in other lines.
As I said, of course I tried hello world on serial and even a program to test the SD card. It works. That' why I jumped to the tft. My programm works on a lolin d32 pro with tft e spi library (m5 stack in the user setup same pinouts). But on this one its not working.
Thanks for your help

**Reply by elgerg** [reply]
Ah, sorry.

It would be interesting to know what board package version you are using. If it's above 2.0.14 you might be having the problem described here:
https://github.com/Bodmer/TFT_eSPI/issues/3384

**Comment by ced31esp** [comment]
Thanks to your advice, my tft is working ! 
I add #define USE_HSPI_PORT in the file Setup70b.h

I changed the version of the board package from the last one to the 2.0.13. (Don't know if it was necessary).

I have one more problem now. It wasn't the case with my lolin d32 pro. I can't have at the same time the sd card and the tft working. I tested my sdcard on a program it's ok, I tested my tft it's ok too. But when I mixed both of them , it can't read something on the sd card and display it.

For example : 


#include <TFT_eSPI.h> // Bibliothèque pour l'écran TFT
#include <SPI.h>      // Bibliothèque pour la communication SPI
#include <SD.h>       // Bibliothèque pour la carte SD

//  SD pin
#define SD_CS 46



TFT_eSPI tft = TFT_eSPI(); // Initialiser l'écran TFT

void setup() {
  Serial.begin(115200);
  tft.begin();
  tft.setRotation(1); // Rotation de l'écran (1 pour portrait)
  tft.fillScreen(TFT_BLACK); // Effacer l'écran
  tft.setTextColor(TFT_WHITE, TFT_BLACK); // Couleur du texte
  tft.setTextSize(2); // Taille du texte
  tft.print("Hello World");

  // Initialiser la carte SD
  if (!SD.begin(SD_CS)) {
    Serial.println("Erreur d'initialisation de la carte SD !");
    return;
  }

  // Ouvrir le fichier texte
  File file = SD.open("/test.txt");
  if (!file) {
    Serial.println("Erreur d'ouverture du fichier !");
    return;
  }

  
  while (file.available()) {
    String name= file.readStringUntil('\n');
    name.trim(); 
    tft.fillScreen(TFT_BLACK); // Effacer l'écran
    tft.setCursor(0, 0); // Positionner le curseur
    tft.println(name);
    delay(1000); 
  }

  file.close();
}

void loop() {
  
}
Hello world is displayed then nothing.

**Reply by elgerg** [reply]
Have a look at this
https://github.com/Bodmer/TFT_eSPI/discussions/2717#discussioncomment-6351532

**Reply by ced31esp** [reply]
Thanks, I have a look to the discussion above. I'd like to try the vspi and hspi but I have difficulties to find the pins on my lolin S3 Pro corrresponding. 

For the TFT I set  #define TFT_CS   48 
#define TFT_MOSI 11 
#define TFT_SCLK 12 
#define TFT_MISO 13
#define TFT_DC    47
#define TFT_RST   21

in the program below from the discussion 2717, what pins others thant tft do I have to put spiSD.begin(18,19,23,5) according to my lolin S3 pro
CS: Pin 
MISO: Pin 
MOSI: Pin 
CLK: Pin   



![Capture d’écran 2024-08-05 à 20 43 05](https://github.com/user-attachments/assets/b38d95ee-dfc9-4ad5-bd0e-8d6074e68409)

SPIClass spiSD(VSPI);
SPIClass spiTFT(HSPI);

#include <TFT_eSPI.h>      // Hardware-specific library

TFT_eSPI tft = TFT_eSPI(); // Invoke custom library

File root;

int fileCountOnSD = 0; // for counting files



void setup(){
    Serial.begin(115200);

spiSD.begin(18,19,23,5);
spiTFT.begin(14,12,13,15);

thanks for your reply

**Comment by ced31esp** [comment]
After a few tries, it seems ok with digitalWrite (SD_CS,HIGH)' same.with tft.
I 'll try more precisely and I 'll tell you.
Thx
